File: app/src/main/java/com/mynet/kazekima/wsfighterscard/RecentResults.java
/*
 * Copyright (c) 2018 Ryuusuke Azuma All Rights Reserved.
 */

package com.mynet.kazekima.wsfighterscard;

import android.database.Cursor;
import android.os.Bundle;
import android.widget.ListView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.loader.app.LoaderManager;
import androidx.loader.content.CursorLoader;
import androidx.loader.content.Loader;

import com.mynet.kazekima.fuse.ActivityBridge;
import com.mynet.kazekima.wsfighterscard.data.FightersDb;

public class RecentResults implements LoaderManager.LoaderCallbacks<Cursor> {

    private RecentResultsListAdapter mAdapter;

    RecentResults() {
        AppCompatActivity activity = ActivityBridge.getInstances().getActivity();
        ListView listView = (ListView) activity.findViewById(R.id.listView);
        mAdapter = new RecentResultsListAdapter(activity, null, true);
        listView.setAdapter(mAdapter);
        mAdapter.notifyDataSetChanged();
        LoaderManager.getInstance(activity).initLoader(0, null, this);
    }

    @NonNull
    @Override
    public Loader<Cursor> onCreateLoader(int i, @Nullable Bundle bundle) {
        AppCompatActivity activity = ActivityBridge.getInstances().getActivity();
        return new CursorLoader(activity, FightersDb.Game.CONTENT_URI,
                null, null, null, null);
    }

    @Override
    public void onLoadFinished(@NonNull Loader<Cursor> loader, Cursor cursor) {
        mAdapter.swapCursor(cursor);
        mAdapter.notifyDataSetChanged();
    }

    @Override
    public void onLoaderReset(@NonNull Loader<Cursor> loader) {
        mAdapter.swapCursor(null);
    }

    public void finish() {
        AppCompatActivity activity = ActivityBridge.getInstances().getActivity();
        if (activity != null) {
            LoaderManager.getInstance(activity).destroyLoader(0);
        }
    }
}

File: android/src/main/kotlin/com/mynet/kazekima/wsfighterscard/RecentResults.kt
/*
 * Copyright (c) 2025 Ryuusuke Azuma All Rights Reserved.
 */

package com.mynet.kazekima.wsfighterscard

import android.database.Cursor
import android.os.Bundle
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import androidx.loader.app.LoaderManager
import androidx.loader.content.CursorLoader
import androidx.loader.content.Loader
import com.mynet.kazekima.wsfighterscard.data.FightersDb

class RecentResults(private val mActivity: AppCompatActivity) : LoaderManager.LoaderCallbacks<Cursor> {
    private val mAdapter: RecentResultsListAdapter

    init {
        val listView = mActivity.findViewById<ListView>(R.id.listView)
        mAdapter = RecentResultsListAdapter(mActivity, null, true)
        listView.adapter = mAdapter
        mAdapter.notifyDataSetChanged()
        LoaderManager.getInstance(mActivity).initLoader(0, null, this)
    }

    override fun onCreateLoader(i: Int, bundle: Bundle?): Loader<Cursor> {
        return CursorLoader(mActivity, FightersDb.Game.CONTENT_URI,
                null, null, null, null)
    }

    override fun onLoadFinished(loader: Loader<Cursor>, cursor: Cursor?) {
        mAdapter.swapCursor(cursor)
        mAdapter.notifyDataSetChanged()
    }

    override fun onLoaderReset(loader: Loader<Cursor>) {
        mAdapter.swapCursor(null)
    }

    fun finish() {
        LoaderManager.getInstance(mActivity).destroyLoader(0)
    }
}
